tensorflow_probability/python/positive_semidefinite_kernels/polynomial_test.py

```diff
@@ -0,0 +1,187 @@
+    k = psd_kernels.Polynomial(
+      bias_variance=0.,
+      slope_variance=1.,
+      exponent=1.,
+      feature_ndims=1)
+    x = np.ones([5, 3], np.float32)
+    y = np.ones([5, 3], np.float32)
+    k.apply(x, y)
+
+  def testFloat64(self):
+    # No exception expected
+    k = psd_kernels.Polynomial(
+      bias_variance=np.float64(0.),
+      slope_variance=np.float64(1.),
+      exponent=np.float64(1.),
+      feature_ndims=1)
+    x = np.ones([5, 3], np.float64)
+    y = np.ones([5, 3], np.float64)
+    k.apply(x, y)
+
+  @parameterized.named_parameters(
+    ('1 feature dimension', 1, (5, 3), (5, 3), (5,)),
+    ('2 feature dimensions', 2, (5, 3, 2), (5, 3, 2), (5,)))
+  def testShapesAreCorrectApply(self, feature_ndims,
+                                x_shape, y_shape, apply_shape):
+    k = psd_kernels.Polynomial(
+      bias_variance=0.,
+      slope_variance=1.,
+      exponent=1.,
+      feature_ndims=feature_ndims)
+    x = np.ones(x_shape, np.float32)
+    y = np.ones(y_shape, np.float32)
+    self.assertAllEqual(
+      k.apply(x, y).shape, apply_shape)
+
+  @parameterized.named_parameters(
+    ('1 feature dimension, 1 batch dimension',
+     1, (5, 3), (4, 3), (5, 4)),
+    ('1 feature dimension, 2 batch dimensions',
+     1, (10, 5, 3), (10, 4, 3), (10, 5, 4)),
+    ('2 feature dimensions, 1 batch dimension',
+     2, (5, 3, 2), (4, 3, 2), (5, 4)))
+  def testShapesAreCorrectMatrix(self, feature_ndims,
+                                 x_shape, y_shape, matrix_shape):
+    k = psd_kernels.Polynomial(
+      bias_variance=0.,
+      slope_variance=1.,
+      exponent=1.,
+      feature_ndims=feature_ndims)
+    x = np.ones(x_shape, np.float32)
+    y = np.ones(y_shape, np.float32)
+    self.assertAllEqual(
+      k.matrix(x, y).shape, matrix_shape)
+
+  def testShapesAreCorrectBroadcast(self):
+    k = psd_kernels.Polynomial(
+      bias_variance=np.ones([2, 1, 1], np.float32),
+      slope_variance=np.ones([1, 3, 1], np.float32))
+    self.assertAllEqual(
+        k.matrix(
+          np.ones([2, 4, 3], np.float32),
+          np.ones([2, 5, 3], np.float32)
+        ).shape, [2, 3, 2, 4, 5])
+    #             `--'  |  `--'
+    #               |   |    `- matrix shape
+    #               |   `- from input batch shapes
+    #               `- from broadcasting kernel params
+
+  def testValuesAreCorrect(self):
+    bias_variance = 1.5
+    slope_variance = 0.5
+    exponent = 2
+    k = psd_kernels.Polynomial(
+      bias_variance=bias_variance,
+      slope_variance=slope_variance,
+      exponent=exponent
+    )
+    x = np.random.uniform(-1, 1, size=[5, 3]).astype(np.float32)
+    y = np.random.uniform(-1, 1, size=[4, 3]).astype(np.float32)
+    self.assertAllClose(
+      self.evaluate(k.matrix(x, y)),
+      bias_variance**2 + slope_variance**2 * (x.dot(y.T))**exponent
+    )
+
+
+@test_util.run_all_in_graph_and_eager_modes
+class LinearTest(tf.test.TestCase, parameterized.TestCase):
+
+  def isPolynomial(self):
```

Comment:
is this meant to be a test method? I believe the test runner will only execute it if the name starts with 'test'

Reply:
I wasn't aware. I updated, tested it, and amended the PR.